x-pack/plugins/alerting/server/alerts_service/lib/set_alerts_to_untracked.ts

```diff
@@ -183,9 +185,13 @@ export async function setAlertsToUntracked({
 }
 
 // Certain rule types don't flatten their AAD values, apply the ALERT_STATUS key to them directly
-const UNTRACK_UPDATE_PAINLESS_SCRIPT = `
+const getUntrackUpdatePainlessScript = (now: Date) => `
 if (!ctx._source.containsKey('${ALERT_STATUS}') || ctx._source['${ALERT_STATUS}'].empty) {
```

Comment:
Why do we need condition here? Can we just use square bracket notation?
P.S. Sorry if a question is silly. First time see painless 
:-)

Reply:
Square bracket notation doesn't work on rule types that don't flatten their Alerts as Data values:
```json
"kibana": {
  "alert": {
    "status": "active"
  }
}
```
 I know the ES Query rule type and the Index Threshold rule type do it this way. We need to fix those rule types at some point, but right now we need this conditional in the script to support all rule types.